cyvcf2/cyvcf2.pyx

```diff
@@ -670,6 +670,41 @@ cdef class VCF(HTSFile):
             stdlib.free(cnames)
             return self._seqnames
 
+    cdef _num_records(self):
+        cdef uint64_t total, records, v;
+        cdef int ret, tid, nseq;
+        cdef hts_idx_t *idx = NULL;
+
+        if self.hidx == NULL and self.idx == NULL:
+            # Try to open an index
+            if self.fname.decode(ENC).endswith(('.bcf', '.bcf.gz')):
+                self.hidx = bcf_index_load(self.fname)
+            else:
+                self.idx = tbx_index_load(to_bytes(self.fname))
+        if self.hidx != NULL:
+            idx = self.hidx
+            assert self.idx == NULL
+        if self.idx != NULL:
+            idx = self.idx.idx
+            assert self.hidx == NULL
+
+        if idx == NULL:
```

Comment:
I thought a bit about this, and since you *need* an index to compute this, it seems best to make it a hard error so that the user is alerted. Happy to discuss of course (also the type of the exception is debateable)

Reply:
This is fine with me. You could just document in `num_records` docstring that is will raise a ValueError if index is not found.

Reply:
done